doc/source/reference/c-api.iterator.rst

```diff
@@ -754,10 +757,19 @@ Construction and Destruction
 
     Returns ``NPY_SUCCEED`` or ``NPY_FAIL``.
 
+.. c:function:: int NpyIter_Close(NpyIter* iter)
+
+    Resolves any needed writeback resolution. Must be called before
+    ``NpyIter_Deallocate``. After this call it is not safe to use the operands.
+
+    Returns ``0`` or ``-1`` if unsuccessful.
+
 .. c:function:: int NpyIter_Deallocate(NpyIter* iter)
 
     Deallocates the iterator object.  This additionally frees any
-    copies made, triggering UPDATEIFCOPY behavior where necessary.
+    copies made. If deallocating triggers WRITEBACKIFCOPY behavior, a
+    deprecation warning will be emitted, this means ``NpyIter_Close`` should
+    have been called preceeding this call.
```

Comment:
I might rephrase:
```
Deallocates the iterator object.

`NpyIter_Close` should be called before this. If not, and if writeback is needed, it will be performed at this point in order to maintain back-compatibility with older code, and a deprecation warning will be emmitted. Old code shuold be updated to call `NpyIter_Close` beforehand.
```

Reply:
fixed